BraintreeCore/src/main/java/com/braintreepayments/api/AuthorizationLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.braintreepayments.api;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+class AuthorizationLoader {
+
+    private Authorization authorization;
+    private final ClientTokenProvider clientTokenProvider;
+
+    AuthorizationLoader(@Nullable String initialAuthString, @Nullable ClientTokenProvider clientTokenProvider) {
+        this.clientTokenProvider = clientTokenProvider;
+        if (initialAuthString != null) {
+            this.authorization = Authorization.fromString(initialAuthString);
+        }
+    }
+
+    void loadAuthorization(@NonNull final AuthorizationCallback callback) {
+        if (authorization != null) {
```

Comment:
Are we planning to keep tuple single method callbacks for internal or do we want this to align more with the `ClientTokenCallback` pattern? Either way seems ok to me just curious about the decision

Reply:
Yeah for here, I was following that one piece of feedback we got for dead code paths. I'm open to either I guess this depends on the overall callback / listener balance in the API. I think the tuple still makes sense for SAM but we could go with listeners everywhere it might not be too bad.